include/aspect/particle/world.h

```diff
@@ -83,6 +83,7 @@ namespace aspect
          * take the properties of the closest particle, but more complicated
          * schemes are possible.
          *
+         * @param [in] generator The particle generator for this world.
          * @param [in] manager The property manager for this world.
          * @param [in] load_balancing The strategy used to balance the
          * computational load for particle advection across processes.
```

Comment:
Yes, I would say the best strategy is to pass down raw pointers (which already indicates that the *calling* side does not manage them via some kind of smart pointer class) and then the *callee* takes over ownership. You would indicate this in the interface documentation of the function: `@note This function takes over ownership of the objects pointed to by @p generator, @p manager, [...complete list...] objects.`.

Which also means that the list of arguments above is not complete or correctly sorted :-)